libs/platform/unit/platform_unit_test.cpp

```diff
@@ -701,3 +702,59 @@ BIT_MASK_TEST(33);
 BIT_MASK_TEST(65);
 BIT_MASK_TEST(129);
 BIT_MASK_TEST(1025);
+
+TEST_CASE("ring_buffer", "[platform]")
+{
+    _test_helper test_helper;
+    size_t consumer;
+    size_t producer;
+    ebpf_ring_buffer_t* ring_buffer;
+
+    uint8_t* buffer;
+    std::vector<uint8_t> data(10);
+    size_t size = 64 * 1024;
+
+    REQUIRE(ebpf_ring_buffer_create(&ring_buffer, size) == EBPF_SUCCESS);
+    REQUIRE(ebpf_ring_buffer_map_buffer(ring_buffer, &buffer) == EBPF_SUCCESS);
+
+    ebpf_ring_buffer_query(ring_buffer, &consumer, &producer);
+
+    // Ring is empty
+    REQUIRE(producer == consumer);
+    REQUIRE(consumer == 0);
+
+    REQUIRE(ebpf_ring_buffer_output(ring_buffer, data.data(), data.size()) == EBPF_SUCCESS);
+    ebpf_ring_buffer_query(ring_buffer, &consumer, &producer);
+
+    // Ring is not empty
+    REQUIRE(producer != consumer);
+    REQUIRE(producer == data.size() + EBPF_OFFSET_OF(ebpf_ring_buffer_record_t, data));
+    REQUIRE(consumer == 0);
+
+    auto record = ebpf_ring_buffer_next_record(buffer, size, consumer, producer);
+    REQUIRE(record != nullptr);
+    REQUIRE(record->header.length == data.size() + EBPF_OFFSET_OF(ebpf_ring_buffer_record_t, data));
+
+    REQUIRE(ebpf_ring_buffer_return(ring_buffer, record->header.length) == EBPF_SUCCESS);
+    ebpf_ring_buffer_query(ring_buffer, &consumer, &producer);
+
+    record = ebpf_ring_buffer_next_record(buffer, size, consumer, producer);
+    REQUIRE(record == nullptr);
+    REQUIRE(consumer == producer);
+    REQUIRE(producer == data.size() + EBPF_OFFSET_OF(ebpf_ring_buffer_record_t, data));
+    REQUIRE(consumer == data.size() + EBPF_OFFSET_OF(ebpf_ring_buffer_record_t, data));
+
+    data.resize(1023);
+    while (ebpf_ring_buffer_output(ring_buffer, data.data(), data.size()) == EBPF_SUCCESS) {
+    }
+
+    ebpf_ring_buffer_query(ring_buffer, &consumer, &producer);
+    REQUIRE(ebpf_ring_buffer_return(ring_buffer, (producer - consumer) % size) == EBPF_SUCCESS);
```

Comment:
I understand that producer/consumer are ever increasing - but since produce cannot write beyond capacity, is the %size needed here?